test/extended/testdata/long_names/fixture.json

```diff
@@ -0,0 +1,91 @@
+{
+    "kind": "List",
+    "apiVersion": "v1",
+    "metadata": {},
+    "items": [
+        {
+            "kind": "ImageStream",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": "long-names-a"
+            }
+        },
+        {
+            "kind": "ImageStream",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": "long-names-b"
+            }
+        },
+        {
+            "kind": "BuildConfig",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": "12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890a",
+                "labels": {
+                    "name": "12345678901234567890123456789012345678901234567890123456789012a"
+                }
+            },
+            "spec": {
+                "triggers": [],
+                "source": {
+                    "type": "Git",
+                    "git": {
+                        "uri": "git://github.com/openshift/ruby-hello-world.git"
+                    }
+                },
+                "strategy": {
+                    "type": "Source",
+                    "env": [],
+                    "sourceStrategy": {
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "centos/ruby-22-centos7"
+                        }
+                    }
+                },
+                "output": {
```

Comment:
you can make this buildconfig execute faster by leaving out the output so it doesn't have to push the resulting image anywhere.